akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -27,17 +27,42 @@ class Encoder(
       private val localAddress = transport.localAddress.address
       private val serialization = SerializationExtension(transport.system)
 
+      private val noSender = transport.system.deadLetters.path.toSerializationFormatWithAddress(localAddress)
+      private val senderCache = new java.util.HashMap[ActorRef, String]
+      private var recipientCache = new java.util.HashMap[ActorRef, String]
+
       override def onPush(): Unit = {
         val send = grab(in)
         val envelope = pool.acquire()
 
-        headerBuilder.recipientActorRef = send.recipient.path.toSerializationFormat
+        val recipientStr = recipientCache.get(send.recipient) match {
+          case null ⇒
+            val s = send.recipient.path.toSerializationFormat
+            // FIXME we might need an efficient LRU cache
```

Comment:
No, this should be handled by the compression table, since in many cases we don't even need the cached string, only an index. The nice thing is that the compression table by definition will store the most common entries anyway, so that will be better than LRU